packages/client/hmi-client/src/components/workflow/tera-workflow.vue

```diff
@@ -465,7 +482,15 @@ function onMenuSelection(operatorType: string, menuNode: WorkflowNode<any>) {
 	newNodePosition.y = menuNode.y;
 
 	if (name && operation && node && drilldown) {
-		addOperatorToWorkflow({ name, operation, node, drilldown })();
+		const newNode: WorkflowNode<any> = addOperatorToWorkflow({ name, operation, node, drilldown })();
+		const inputPort = newNode.inputs.find((input) => input.type.includes(port.type));
+
+		if (inputPort) {
+			// wait for the DOM to load the new node before adding the edge
+			await nextTick();
+			createNewEdge(menuNode, port, direction);
+			createNewEdge(newNode, inputPort, 0);
```

Comment:
`createNewEdge` is meant for interactive use - when modellers are dragging stuff. 

If you want to "auto connect" use `wf.value.addEdge(...)` instead, it is more straightforward and you don't need to deal with "newEdge" transient state.